### ISSUE TITLE
5.0.7 TMDB Timezone Assumptions

### ISSUE THREAD

**Issue by TommySnyder** [TV-Rename/tvrename#983]
Starting at https://groups.google.com/g/tvrename/c/3as3cBh2haM/m/4NqnQh_UAQAJ

Sparky53
Jul 12, 2023, 9:53:29 PM (13 days ago) 
to TV Rename
OK, few things here.

1) Try a force refresh all to get the latest upstream data.

A) If we get a time from the source then we use that. (TMDB does not provide a time). If not we use 8pm for non-streaming networks and midnight for streaming networks.  (streaming defined here: https://github.com/TV-Rename/tvrename/blob/10450861d46dab236f2362141112fc3b812d3d9d/TVRename/Utility/Helper/TimeZoneHelper.cs#L32)
B) These are the 'local' time and converted into the users time based on the user's timezone and the timezone of the series (on the edit show screen)

So, in thoery, this show should default to 8pm EST. If you want them at 2pm then you need to put a 6hr offset in the timezone field (to GMT-2).

Tom Snyder
Jul 13, 2023, 4:13:56 AM (13 days ago) 
to TV Rename
Looking at A) in your message, it lists the TV network the show is on as ABC (US) in your code but in TV Rename (see my screenshot above) merely as ABC.  So, I don't think it is picking it up as a network then and so the 6hr offset doesn't work.  It won't pick it up as a streamer either so I have no idea what it defaults to.
Although presently all the American Broadcast Company shows on my screen are listed as simply ABC, I have seen others listed as ABC (US) such as Grey's Anatomy.  And obviously they have nothing to do with ABC (AU) in Australia.
And just as a point of interest, all the ABC/ABC (US) shows are later streamed on Hulu.  The show I'm having problems with, General Hospital, broadcasts at 2pm and then streams on Hulu at 8pm.  For the most part, all the prime time ABC/ABC (US) shows are streamed on Hulu at 5am.  All the times I've listed are ET (Eastern Time, Eastern Standard Time in Winter or Eastern Daylight Savings Time in Summer)

Summary:  ABC needs to be added to the list of US Networks found in https://github.com/TV-Rename/tvrename/blob/10450861d46dab236f2362141112fc3b812d3d9d/TVRename/Utility/Helper/TimeZoneHelper.cs#L32)
Also, the list of streamers is a little weak. Hulu, Peacock, Discovery+ and others could be added to it.  That's not even counting the European, Australian and Asian streamers.

**Comment by SirSparkles** [comment]
I have added what you asked for, but I really don't think it will help you.

TimeZoneForNetwork is only used to create a guess for the show when we are adding a show with no human input.

Adding those 3 streamers will not impact your ABC show

**Comment by TommySnyder** [comment]
Fingers crossed anyway.  I'll let you know after 5.0.8.  Thank you so much.

**Comment by TommySnyder** [comment]
﻿I think you fixed it. I usually use TVDB for TV shows but they have the episode numbers wrong for two of my wife's soap operas and so I have to use TMDB. As you explained, TMDB doesn't keep track of air time. General Hospital is a network show on ABC (not listed as ABC (US) by TMDB) and Days of our Lives is a streamer on Peacock (formerly on NBC network). Anyway, after your fix I was able to easily set them to their correct time of air or stream. Previously, I believe, I had to set General Hospital to "etc/GMT+18" which is impossible to do. 
As for the three streamer names, they are ones that I use frequently. Having seen them missing from your code, I suggested adding them without understanding the significance of them being there. Obviously, feel free to remove them if you think they might do some harm.
As always, thank you, thank you, thank you.
